domain/secret/service/service.go

```diff
@@ -525,7 +531,20 @@ func splitCharmSecretOwners(owners ...CharmSecretOwner) (domainsecret.Applicatio
 // The count of secret and revisions in the result must match.
 func (s *SecretService) ListCharmSecrets(ctx context.Context, owners ...CharmSecretOwner) ([]*secrets.SecretMetadata, [][]*secrets.SecretRevisionMetadata, error) {
 	appOwners, unitOwners := splitCharmSecretOwners(owners...)
-	return s.secretState.ListCharmSecrets(ctx, appOwners, unitOwners)
+
+	var (
+		mds  []*secrets.SecretMetadata
+		revs [][]*secrets.SecretRevisionMetadata
+	)
+	err := s.secretState.RunAtomic(ctx, func(ctx domain.AtomicContext) error {
```

Comment:
A top level service method which just lists charm secrets should not need run atomic. 
If there's another workflow which does use it in the context of performing updates, that needs to use a bespoke state method.
eg One use case is just to get a Label to add to consumer info after loading the consumer info. This logic could be added to the GetConsumerInfo() state api.
The other use case is to select a single metadata record matching a uri or label. This also should be a bespoke state method.